### DISCUSSION TITLE
Using read replica in bull

### DISCUSSION THREAD

**Discussion by alko89** [taskforcesh/bullmq#1487] [Q&A] [answered]
Is it possible to configure bull to use read replicas when reading jobs? Basically to use two connections, one for scheduling jobs and another to fetch them?

**Comment by manast** [comment, marked as answer]
When you fetch you are also updating the queue, i.e. writing. So the answer is no. Only the getters would make sense in a read-only replica.

**Reply by alko89** [reply]
Ok so having read replicas for Bull doesn't make much sense. Better to split jobs into its own redis instances with no replicas if we want to improve performance?

**Reply by manast** [reply]
Replicas make sense for failover redundancy, as they have it implemented in AWS elasticache. Otherwise there is no performance benefit. Do you have any performance issues?

**Reply by alko89** [reply]
We're still investigating, now it seems to be a fault on our side. But thanks for your answers, they were very helpful!

**Reply by manast** [reply]
Unless the CPU in the Redis instance is over 80% then the bottleneck is somewhere else.

**Reply by alko89** [reply]
Ohh its at a consistent 100% :smile: but we're working on some optimizations in code. For the future we might also consider either using a sharded redis instance or having a tiny redis for each job.

**Reply by DanielNetzer** [reply]
interesting this is exactly what we are facing...

**Reply by alko89** [reply]
I've actually fixed it, there was a bug in the way we scheduled repeatable jobs. After the fix the usage dropped 70%

**Reply by DanielNetzer** [reply]
very interesting, do you mind sharing what was the issue and how did you fixed it?

**Reply by alko89** [reply]
It was actually a very specific issue, the dev that wrote the code was checking that the job is not already scheduled by basically reading all jobs before deciding to create a new one. This can easily be fixed by setting a jobId manually which ensures the job with the same id (and in case of a scheduled job, the same `pattern`) is not scheduled twice.

**Reply by DanielNetzer** [reply]
I see, yea make sense.
